netobserv/v1alpha1/flowcollector_types.go

```diff
@@ -0,0 +1,436 @@
+
+	//+kubebuilder:validation:Minimum=0
+	//+kubebuilder:default:=1
+	// Replicas defines the number of replicas (pods) to start for Deployment kind. Ignored for DaemonSet.
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// HPA spec of an horizontal pod autoscaler to set up for the collector Deployment. Ignored for DaemonSet.
+	// +optional
+	HPA *FlowCollectorHPA `json:"hpa,omitempty"`
+
+	//+kubebuilder:validation:Minimum=1025
+	//+kubebuilder:validation:Maximum=65535
+	//+kubebuilder:default:=2055
+	// Port is the collector port: either a service port for Deployment kind, or host port for DaemonSet kind
+	// By conventions, some value are not authorized port must not be below 1024 and must not equal this values:
+	// 4789,6081,500, and 4500
+	Port int32 `json:"port,omitempty"`
+
+	//+kubebuilder:validation:Minimum=1
+	//+kubebuilder:validation:Maximum=65535
+	//+kubebuilder:default:=8080
+	// HealthPort is a collector HTTP port in the Pod that exposes the health check API
+	HealthPort int32 `json:"healthPort,omitempty"`
+
+	//+kubebuilder:validation:Minimum=1
+	//+kubebuilder:validation:Maximum=65535
+	//+kubebuilder:default:=9090
+	// PrometheusPort is the prometheus HTTP port: this port exposes prometheus metrics
+	PrometheusPort int32 `json:"prometheusPort,omitempty"`
+
+	//+kubebuilder:default:="quay.io/netobserv/flowlogs-pipeline:main"
+	// Image is the collector image (including domain and tag)
+	Image string `json:"image,omitempty"`
+
+	//+kubebuilder:validation:Enum=IfNotPresent;Always;Never
+	//+kubebuilder:default:=IfNotPresent
+	// ImagePullPolicy is the Kubernetes pull policy for the image defined above
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
+
+	//+kubebuilder:validation:Enum=trace;debug;info;warn;error;fatal;panic
+	//+kubebuilder:default:=info
+	// LogLevel defines the log level for the collector runtime
+	LogLevel string `json:"logLevel,omitempty"`
+
+	//+kubebuilder:default:={requests:{memory:"100Mi",cpu:"100m"},limits:{memory:"300Mi"}}
+	// Compute Resources required by this container.
+	// Cannot be updated.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,8,opt,name=resources"`
+
+	//+kubebuilder:default:=true
+	// EnableKubeProbes is a flag to enable or disable Kubernetes liveness/readiness probes
+	EnableKubeProbes bool `json:"enableKubeProbes,omitempty"`
+}
+
+type FlowCollectorHPA struct {
+	// minReplicas is the lower limit for the number of replicas to which the autoscaler
+	// can scale down.  It defaults to 1 pod.  minReplicas is allowed to be 0 if the
+	// alpha feature gate HPAScaleToZero is enabled and at least one Object or External
+	// metric is configured.  Scaling is active as long as at least one metric value is
+	// available.
+	// +optional
+	MinReplicas *int32 `json:"minReplicas,omitempty" protobuf:"varint,2,opt,name=minReplicas"`
```

Comment:
Why is this a pointer, if maxreplicas isn't?

Reply:
I'm pretty sure there was a reason, good or bad ... @OlivierCazade or @jpinsonneau do you remember?

Reply:
I don't remember, sorry.

Reply:
this directly come from k8s api [HorizontalPodAutoscalerSpec](https://pkg.go.dev/k8s.io/api/autoscaling/v1#HorizontalPodAutoscalerSpec)

Reply:
Ok so, that's then because of 
```
minReplicas is allowed to be 0 if the alpha feature gate HPAScaleToZero is enabled and at least one Object or External metric is configured.
```